queries.sql

```diff
@@ -17,3 +17,65 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
 SELECT * FROM animals WHERE weight_kg Between 10.4 AND 17.3;
+
+BEGIN;
+
+    UPDATE animals
+    SET species = 'unspecified';
+
+    SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+BEGIN;
+
+    DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+BEGIN;
+
+    DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+    SAVEPOINT SP1;
+
+    UPDATE animals 
+    SET weight_kg = weight_kg * -1;
+
+    ROLLBACK TO SP1;
+
+    UPDATE animals 
+    SET weight_kg = weight_kg * -1
+    WHERE weight_kg < 0;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT MAX(escape_attempts), neutered 
+FROM animals
+GROUP BY neutered;
+
+SELECT MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals;
```

Comment:
- [ ] Good job on this query, however, according to the requirements you should return the min and max weight of each type of animal instead of the min and max weight of all animals, which is what this query is returning.

Reply:
@mikethreels 
Can you be more specific about the type? 
Which type are you talking about here?
Is it "species"?